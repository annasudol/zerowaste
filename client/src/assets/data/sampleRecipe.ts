export const sampleRecipe = {
    title: 'Apple Pie',
    servings: 8,
    readyInMinutes: 130,
    ingredients: ['sugar', 'brown sugar', 'flour', 'cinnamon sugar', 'ginger', 'apple'],
    detailedIngredients: ['1/2 cup sugar', '1/2 cup packed brown sugar', '3 tablespoons all-purpose flour', '1 teaspoon ground cinnamon', '1/4 teaspoon ground ginger', '1/4 teaspoon ground nutmeg', '6 to 7 cups thinly sliced peeled tart apples', '1 tablespoon lemon juice', 'Pastry for double-crust pie', '1 tablespoon butter'],
    instructions: 'In a small bowl, combine the sugars, flour and spices; set aside. In a large bowl, toss apples with lemon juice. Add sugar mixture; toss to coat.\nPie plate with bottom crust; trim even with edge. Fill with apple mixture; dot with butter. Roll remaining crust to fit the top of pie; place overfilling. Trim, seal and flute edges. Cut slits in crust.\nBeat egg white until foamy; brush over crust. Sprinkle with sugar. Cover edges loosely with foil.\nBake at 375° for 25 minutes. Remove foil and bake until crust is golden brown and filling is bubbly, 20-25 minutes longer. Cool on a wire rack.', image: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFYAAABWCAYAAABVVmH3AAAMSGlDQ1BJQ0MgUHJvZmlsZQAASImVVwdYU8kWnltSSWiBCEgJvQkiSJcSQosgIFWwEZJAQokhIYjYlWUVXLuIgA1dFVF0dQVkrahrXQR7f1hQUdbFgg2VNymg637vve+d75t7/5w55z8lc++dAUCnhieV5qK6AORJCmTxESGsCalpLNIjgAMMaAEjYMTjy6XsuLhoAGXw/nd5ew0gyvtlFyXXP+f/q+gJhHI+AEgcxBkCOT8P4l8BwEv4UlkBAEQfqLeeXiBV4kkQG8hgghBLlThLjUuUOEONK1U2ifEciHcBQKbxeLIsALSboZ5VyM+CPNo3IHaTCMQSAHTIEAfyRTwBxJEQj8jLm6bE0A44ZHzDk/U3zowhTh4vawira1EJOVQsl+byZvyf7fjfkperGIxhBwdNJIuMV9YM+3YjZ1qUEtMg7pFkxMRCrA/xe7FAZQ8xShUpIpPU9qgpX86BPQNMiN0EvNAoiE0hDpfkxkRr9BmZ4nAuxHCFoEXiAm6ixneRUB6WoOGskU2Ljx3EmTIOW+PbwJOp4irtTypyktga/hsiIXeQ/02xKDFFnTNGLRQnx0CsDTFTnpMQpbbBbIpFnJhBG5kiXpm/DcR+QklEiJofm5IpC4/X2Mvy5IP1YotEYm6MBlcViBIjNTy7+DxV/kYQNwsl7KRBHqF8QvRgLQJhaJi6dqxdKEnS1It1SgtC4jW+r6S5cRp7nCrMjVDqrSA2lRcmaHzxwAK4INX8eIy0IC5RnSeekc0bG6fOBy8C0YADQgELKODIANNANhC39TT1wF/qmXDAAzKQBYTARaMZ9EhRzUjgNQEUgz8hEgL5kF+IalYICqH+85BWfXUBmarZQpVHDngMcR6IArnwt0LlJRmKlgweQY34H9H5MNdcOJRz/9SxoSZao1EM8rJ0Bi2JYcRQYi…',
    sourceUrl: 'myblog.com'
};