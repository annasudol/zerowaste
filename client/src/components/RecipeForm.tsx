import { Form, Input, Button, InputNumber } from 'antd';
const { TextArea } = Input;
import { FormInstance } from 'antd/lib/form';
import { AutoComplete, List, ListInput, ImageUpload, FileType } from '../components'
import * as React from 'react';
import { RecipeStateProps } from '../utils/types';
import { Store } from 'antd/lib/form/interface';
import { sampleRecipe } from '../assets/data/sampleRecipe'
const tailLayout = { wrapperCol: { offset: 10, span: 12 } };
const formRef = React.createRef<FormInstance>();

import * as request from 'superagent';
import { handlePhotoDelete } from "../utils/handlePhotoDelete";

const cloudName = "drgb4slzt";
const uploadPreset = "iiiutyfi";

const validateMessages = {
    required: '${label} is required!',
    number: {
        range: '${label} must be between ${min} and ${max}',
    },
};


interface RecipeFormProps {
    handleSubmit(inputValues: Store, imageUrl: string): void
    fillForm?: boolean
    defaultValues?: RecipeStateProps
}

export const RecipeForm: React.FC<RecipeFormProps> = ({ handleSubmit, fillForm = false, defaultValues }): React.ReactElement => {
    const [ingredients, setIngredients] = React.useState<string[] | []>([])
    const [detailedIngredients, setDetailedIngredients] = React.useState<string[] | []>([])
    const [imageUrl, setImageUrl] = React.useState<string | undefined>(defaultValues?.image);
    const [error, setError] = React.useState<boolean>(false)
    React.useEffect(() => {
        if (defaultValues && formRef.current) {
            formRef.current.setFieldsValue({ ...defaultValues });
            setIngredients(defaultValues.ingredients)
            setDetailedIngredients(defaultValues.detailedIngredients)
        }
    }, [defaultValues, formRef.current]);




    const onFinish = async (values: Store) => {
        const url = `https://api.cloudinary.com/v1_1/drgb4slzt/upload`;
        await request.post(url)
            .field('upload_preset', uploadPreset)
            .field('file', values.image)
            .field('multiple', false)
            .end((error, response) => {
                if (response.ok) {
                    if (imageUrl && response.body.url !== imageUrl) {
                        handlePhotoDelete(imageUrl)
                    }
                    handleSubmit(values, response.body.url)
                }
            });

    };

    const onReset = () => {
        setDetailedIngredients([]);
        setIngredients([]);

        setError(false);
        formRef.current?.resetFields();
    };

    const onFill = () => {
        formRef.current?.setFieldsValue({ ...sampleRecipe });
        setError(false);
        setIngredients(sampleRecipe.ingredients)
        setDetailedIngredients(sampleRecipe.detailedIngredients)
    }


    const deleteDetailedIngredients = (text: string) => {
        const newList = detailedIngredients.filter(ingredient => ingredient !== text)
        setDetailedIngredients([...newList])
    };

    const onFinishFailed = errorInfo => {
        if (errorInfo.errorFields[3].errors) {
            return setError(true);
        }
    };

    return (
        <Form labelCol={{ span: 10 }}
            wrapperCol={{ span: 20 }}
            onFinish={onFinish}
            layout='horizontal' ref={formRef} name='control-ref' onFinishFailed={onFinishFailed} validateMessages={validateMessages}>
            <Form.Item name='title' label='Title' rules={[{ required: true, min: 3 }]}>
                <Input />
            </Form.Item>
            <Form.Item name='servings' label='Servings' rules={[{ required: true, type: 'number', min: 1, max: 10 }]}>
                <InputNumber />
            </Form.Item>
            <Form.Item name='readyInMinutes' label='Ready in Min.' rules={[{ required: true, type: 'number', min: 5, max: 400 }]}>
                <InputNumber />
            </Form.Item>
            <Form.Item label='Ingredients' name='ingredients' rules={[{ required: true, type: 'array', min: 1, max: 20 }]}>
                <AutoComplete error={error} list={ingredients} saveList={(value: string[]): void => setIngredients(value)} form={(value: string[]): void => formRef.current?.setFieldsValue({ ingredients: value })} />
            </Form.Item>
            <Form.Item label='Detailed Ingredients' name='detailedIngredients' rules={[{ required: true, type: 'array', min: 1, max: 10 }]}>
                <ListInput detailedIngredients={detailedIngredients} setDetailedIngredients={(value: string[]): void => setDetailedIngredients(value)} form={(value: string[]): void => formRef.current?.setFieldsValue({ detailedIngredients: value })} />
                <List list={detailedIngredients} deleteItem={(text: string): any => deleteDetailedIngredients(text)} />
            </Form.Item>
            <Form.Item name='instructions' label='Instructions' rules={[{ required: true, min: 10 }]}>
                <TextArea />
            </Form.Item>
            <Form.Item name='image' label='Image' rules={[{ required: false }]}>
                <ImageUpload imageUrl={imageUrl} form={(value: FileType): void => formRef.current?.setFieldsValue({ image: value })} />
            </Form.Item>
            <Form.Item name='sourceUrl' label='Source Url' rules={[{ required: true }]}>
                <Input addonBefore='http://' defaultValue='myblog' />
            </Form.Item>
            <Form.Item {...tailLayout}>
                <Button type='primary' htmlType='submit' className='mr-2'>
                    Submit
                </Button>
                <Button htmlType='button' onClick={onReset} className='mr-2'>
                    Reset
                </Button>
                {fillForm && (<Button type='dashed' htmlType='button' onClick={onFill}>
                    Fill form
                </Button>)}
            </Form.Item>
        </Form>
    );
}
