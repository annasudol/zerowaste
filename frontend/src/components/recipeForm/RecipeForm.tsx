import React, { useState, ReactElement, FC } from 'react';
import { Form, Input, Button, InputNumber } from 'antd';
import { FormInstance } from 'antd/lib/form';
import { AutoComplete, List, ListInput, ImageUpload, FileType } from '..'
import { RecipeStateProps } from '../../utils/types';
import { Store } from 'antd/lib/form/interface';
import { sampleRecipe } from '../../assets/data/sampleRecipe'

import * as request from 'superagent';
import { handlePhotoDelete } from "../../utils/handlePhotoDelete";
const { TextArea } = Input;
const tailLayout = { wrapperCol: { offset: 10, span: 12 } };
const formRef = React.createRef<FormInstance>();


const validateMessages = {
    // eslint-disable-next-line no-template-curly-in-string
    required: '${label} is required!',
    number: {
        // eslint-disable-next-line no-template-curly-in-string
        range: '${label} must be between ${min} and ${max}',
    },
};


interface RecipeFormProps {
    handleSubmit(inputValues: Store, imageUrl: string): void
    fillForm?: boolean
    initialValues?: RecipeStateProps
}

export const RecipeForm: FC<RecipeFormProps> = ({ handleSubmit, fillForm = false, initialValues }): ReactElement => {
    const [ingredients, setIngredients] = React.useState<string[] | []>([])
    const [detailedIngredients, setDetailedIngredients] = React.useState<string[] | []>([])
    const [imageUrl, setImageUrl] = useState<string | undefined>(initialValues?.image);
    const [error, setError] = React.useState<boolean>(false)
    React.useEffect(() => {
        if (initialValues && formRef.current) {
            formRef.current.setFieldsValue({ ...initialValues });
            setIngredients(initialValues.ingredients)
            setDetailedIngredients(initialValues.detailedIngredients)
        }
    }, [initialValues]);




    const onFinish = async (values: Store) => {
        const uploadPreset = process.env.REACT_APP_CLOUD_PRESET;
        const cloudName= process.env.REACT_APP_CLOUD_NAME;
        const url = await `https://api.cloudinary.com/v1_1/${cloudName}/upload`;
        uploadPreset && await request.post(url)
            .field('upload_preset', uploadPreset)
            .field('file', values.image)
            .field('multiple', false)
            .end((_error: any, response: any) => {
                if (response.ok) {
                    if (imageUrl && response.body.url !== imageUrl) {
                        handlePhotoDelete(imageUrl)
                    }
                    return handleSubmit(values, response.body.url)
                }
            });

    
    };

    const onReset = () => {
        setDetailedIngredients([]);
        setIngredients([]);

        setError(false);
        formRef.current?.resetFields();
    };

    const onFill = () => {
        formRef.current?.setFieldsValue({ ...sampleRecipe });
        setError(false);
        setIngredients(sampleRecipe.ingredients)
        setDetailedIngredients(sampleRecipe.detailedIngredients)
    }


    const deleteDetailedIngredients = (text: string) => {
        const newList = detailedIngredients.filter(ingredient => ingredient !== text)
        setDetailedIngredients([...newList])
    };
    const onFinishFailed = (errorInfo: { errorFields: { errors: unknown; }[]; }) => {
        if (errorInfo?.errorFields[3]?.errors) {
            return setError(true);
        }
    };

    return (
        <Form labelCol={{ span: 10 }}
            wrapperCol={{ span: 20 }}
            onFinish={onFinish}
            layout='horizontal' ref={formRef} name='control-ref' onFinishFailed={onFinishFailed} validateMessages={validateMessages}>
            <Form.Item name='title' label='Title' rules={[{ required: true, min: 3 }]}>
                <Input />
            </Form.Item>
            <Form.Item name='servings' label='Servings' rules={[{ required: true, type: 'number', min: 1, max: 10 }]}>
                <InputNumber />
            </Form.Item>
            <Form.Item name='readyInMinutes' label='Ready in Min.' rules={[{ required: true, type: 'number', min: 5, max: 400 }]}>
                <InputNumber />
            </Form.Item>
            <Form.Item label='Ingredients' name='ingredients' rules={[{ required: true, type: 'array', min: 1, max: 20 }]}>
                <AutoComplete error={error} list={ingredients} saveList={(value: string[]): void => setIngredients(value)} form={(value: string[]): void => formRef.current?.setFieldsValue({ ingredients: value })} />
            </Form.Item>
            <Form.Item label='Detailed Ingredients' name='detailedIngredients' rules={[{ required: true, type: 'array', min: 1, max: 10 }]}>
                <ListInput detailedIngredients={detailedIngredients} setDetailedIngredients={(value: string[]): void => setDetailedIngredients(value)} form={(value: string[]): void => formRef.current?.setFieldsValue({ detailedIngredients: value })} />
                <List list={detailedIngredients} deleteItem={(text: string): void => deleteDetailedIngredients(text)} />
            </Form.Item>
            <Form.Item name='instructions' label='Instructions' rules={[{ required: true, min: 10 }]}>
                <TextArea />
            </Form.Item>
            <Form.Item name='image' label='Image' rules={[{ required: true }]}>
                <ImageUpload imageUrl={imageUrl} form={(value: FileType | Blob): void => formRef.current?.setFieldsValue({ image: value })} />
            </Form.Item>
            <Form.Item name='sourceUrl' label='Source Url' rules={[{ required: false }]}>
                <Input addonBefore='http://' />
            </Form.Item>
            <Form.Item {...tailLayout}>
                <Button type='primary' htmlType='submit' className='mr-2'>
                    Submit
                </Button>
                <Button htmlType='button' onClick={onReset} className='mr-2'>
                    Reset
                </Button>
                {fillForm && (<Button type='dashed' htmlType='button' onClick={onFill}>
                    Fill form
                </Button>)}
            </Form.Item>
        </Form>
    );
}
