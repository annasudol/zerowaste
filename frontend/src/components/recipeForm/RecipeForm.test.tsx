import { shallow } from 'enzyme';
import React from 'react';
import { RecipeForm } from './RecipeForm';
import toJson from 'enzyme-to-json';
import { sampleRecipe } from '../../assets/data/sampleRecipe'


describe('component', (): void => {
  describe('LoginSignUpForm', (): void => {
    it('match snapshot', (): void => {
      const wrapper = shallow(<RecipeForm handleSubmit={jest.fn()}/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('match snapshot when initialValues props are given', ()=> {
        const wrapper = shallow(<RecipeForm handleSubmit={jest.fn()} initialValues={sampleRecipe}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

  });


});