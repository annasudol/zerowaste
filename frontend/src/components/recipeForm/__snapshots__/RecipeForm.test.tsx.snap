// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component LoginSignUpForm  1`] = `
<ForwardRef(InternalForm)
  labelCol={
    Object {
      "span": 10,
    }
  }
  layout="horizontal"
  name="control-ref"
  onFinish={[Function]}
  onFinishFailed={[Function]}
  validateMessages={
    Object {
      "number": Object {
        "range": "\${label} must be between \${min} and \${max}",
      },
      "required": "\${label} is required!",
    }
  }
  wrapperCol={
    Object {
      "span": 20,
    }
  }
>
  <FormItem
    label="Title"
    name="title"
    rules={
      Array [
        Object {
          "min": 3,
          "required": true,
        },
      ]
    }
  >
    <Input
      type="text"
    />
  </FormItem>
  <FormItem
    label="Servings"
    name="servings"
    rules={
      Array [
        Object {
          "max": 10,
          "min": 1,
          "required": true,
          "type": "number",
        },
      ]
    }
  >
    <ForwardRef
      step={1}
    />
  </FormItem>
  <FormItem
    label="Ready in Min."
    name="readyInMinutes"
    rules={
      Array [
        Object {
          "max": 400,
          "min": 5,
          "required": true,
          "type": "number",
        },
      ]
    }
  >
    <ForwardRef
      step={1}
    />
  </FormItem>
  <FormItem
    label="Ingredients"
    name="ingredients"
    rules={
      Array [
        Object {
          "max": 20,
          "min": 1,
          "required": true,
          "type": "array",
        },
      ]
    }
  >
    <AutoCompleteForm
      error={false}
      form={[Function]}
      list={Array []}
      saveList={[Function]}
    />
  </FormItem>
  <FormItem
    label="Detailed Ingredients"
    name="detailedIngredients"
    rules={
      Array [
        Object {
          "max": 10,
          "min": 1,
          "required": true,
          "type": "array",
        },
      ]
    }
  >
    <ListInput
      detailedIngredients={Array []}
      form={[Function]}
      setDetailedIngredients={[Function]}
    />
    <List
      deleteItem={[Function]}
      list={Array []}
    />
  </FormItem>
  <FormItem
    label="Instructions"
    name="instructions"
    rules={
      Array [
        Object {
          "min": 10,
          "required": true,
        },
      ]
    }
  >
    <TextArea />
  </FormItem>
  <FormItem
    label="Image"
    name="image"
    rules={
      Array [
        Object {
          "required": true,
        },
      ]
    }
  >
    <ImageUpload
      form={[Function]}
      imageUrl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFYAAABWCAYAAABVVmH3AAAMSGlDQ1BJQ0MgUHJvZmlsZQAASImVVwdYU8kWnltSSWiBCEgJvQkiSJcSQosgIFWwEZJAQokhIYjYlWUVXLuIgA1dFVF0dQVkrahrXQR7f1hQUdbFgg2VNymg637vve+d75t7/5w55z8lc++dAUCnhieV5qK6AORJCmTxESGsCalpLNIjgAMMaAEjYMTjy6XsuLhoAGXw/nd5ew0gyvtlFyXXP+f/q+gJhHI+AEgcxBkCOT8P4l8BwEv4UlkBAEQfqLeeXiBV4kkQG8hgghBLlThLjUuUOEONK1U2ifEciHcBQKbxeLIsALSboZ5VyM+CPNo3IHaTCMQSAHTIEAfyRTwBxJEQj8jLm6bE0A44ZHzDk/U3zowhTh4vawira1EJOVQsl+byZvyf7fjfkperGIxhBwdNJIuMV9YM+3YjZ1qUEtMg7pFkxMRCrA/xe7FAZQ8xShUpIpPU9qgpX86BPQNMiN0EvNAoiE0hDpfkxkRr9BmZ4nAuxHCFoEXiAm6ixneRUB6WoOGskU2Ljx3EmTIOW+PbwJOp4irtTypyktga/hsiIXeQ/02xKDFFnTNGLRQnx0CsDTFTnpMQpbbBbIpFnJhBG5kiXpm/DcR+QklEiJofm5IpC4/X2Mvy5IP1YotEYm6MBlcViBIjNTy7+DxV/kYQNwsl7KRBHqF8QvRgLQJhaJi6dqxdKEnS1It1SgtC4jW+r6S5cRp7nCrMjVDqrSA2lRcmaHzxwAK4INX8eIy0IC5RnSeekc0bG6fOBy8C0YADQgELKODIANNANhC39TT1wF/qmXDAAzKQBYTARaMZ9EhRzUjgNQEUgz8hEgL5kF+IalYICqH+85BWfXUBmarZQpVHDngMcR6IArnwt0LlJRmKlgweQY34H9H5MNdcOJRz/9SxoSZao1EM8rJ0Bi2JYcRQYiâ€¦"
    />
  </FormItem>
  <FormItem
    label="Source Url"
    name="sourceUrl"
    rules={
      Array [
        Object {
          "required": false,
        },
      ]
    }
  >
    <Input
      addonBefore="http://"
      type="text"
    />
  </FormItem>
  <FormItem
    wrapperCol={
      Object {
        "offset": 10,
        "span": 12,
      }
    }
  >
    <Button
      block={false}
      className="mr-2"
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Submit
    </Button>
    <Button
      block={false}
      className="mr-2"
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
    >
      Reset
    </Button>
  </FormItem>
</ForwardRef(InternalForm)>
`;

exports[`component LoginSignUpForm match snapshot 1`] = `
<ForwardRef(InternalForm)
  labelCol={
    Object {
      "span": 10,
    }
  }
  layout="horizontal"
  name="control-ref"
  onFinish={[Function]}
  onFinishFailed={[Function]}
  validateMessages={
    Object {
      "number": Object {
        "range": "\${label} must be between \${min} and \${max}",
      },
      "required": "\${label} is required!",
    }
  }
  wrapperCol={
    Object {
      "span": 20,
    }
  }
>
  <FormItem
    label="Title"
    name="title"
    rules={
      Array [
        Object {
          "min": 3,
          "required": true,
        },
      ]
    }
  >
    <Input
      type="text"
    />
  </FormItem>
  <FormItem
    label="Servings"
    name="servings"
    rules={
      Array [
        Object {
          "max": 10,
          "min": 1,
          "required": true,
          "type": "number",
        },
      ]
    }
  >
    <ForwardRef
      step={1}
    />
  </FormItem>
  <FormItem
    label="Ready in Min."
    name="readyInMinutes"
    rules={
      Array [
        Object {
          "max": 400,
          "min": 5,
          "required": true,
          "type": "number",
        },
      ]
    }
  >
    <ForwardRef
      step={1}
    />
  </FormItem>
  <FormItem
    label="Ingredients"
    name="ingredients"
    rules={
      Array [
        Object {
          "max": 20,
          "min": 1,
          "required": true,
          "type": "array",
        },
      ]
    }
  >
    <AutoCompleteForm
      error={false}
      form={[Function]}
      list={Array []}
      saveList={[Function]}
    />
  </FormItem>
  <FormItem
    label="Detailed Ingredients"
    name="detailedIngredients"
    rules={
      Array [
        Object {
          "max": 10,
          "min": 1,
          "required": true,
          "type": "array",
        },
      ]
    }
  >
    <ListInput
      detailedIngredients={Array []}
      form={[Function]}
      setDetailedIngredients={[Function]}
    />
    <List
      deleteItem={[Function]}
      list={Array []}
    />
  </FormItem>
  <FormItem
    label="Instructions"
    name="instructions"
    rules={
      Array [
        Object {
          "min": 10,
          "required": true,
        },
      ]
    }
  >
    <TextArea />
  </FormItem>
  <FormItem
    label="Image"
    name="image"
    rules={
      Array [
        Object {
          "required": true,
        },
      ]
    }
  >
    <ImageUpload
      form={[Function]}
    />
  </FormItem>
  <FormItem
    label="Source Url"
    name="sourceUrl"
    rules={
      Array [
        Object {
          "required": false,
        },
      ]
    }
  >
    <Input
      addonBefore="http://"
      type="text"
    />
  </FormItem>
  <FormItem
    wrapperCol={
      Object {
        "offset": 10,
        "span": 12,
      }
    }
  >
    <Button
      block={false}
      className="mr-2"
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Submit
    </Button>
    <Button
      block={false}
      className="mr-2"
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
    >
      Reset
    </Button>
  </FormItem>
</ForwardRef(InternalForm)>
`;

exports[`component LoginSignUpForm match snapshot when initialValues props are given 1`] = `
<ForwardRef(InternalForm)
  labelCol={
    Object {
      "span": 10,
    }
  }
  layout="horizontal"
  name="control-ref"
  onFinish={[Function]}
  onFinishFailed={[Function]}
  validateMessages={
    Object {
      "number": Object {
        "range": "\${label} must be between \${min} and \${max}",
      },
      "required": "\${label} is required!",
    }
  }
  wrapperCol={
    Object {
      "span": 20,
    }
  }
>
  <FormItem
    label="Title"
    name="title"
    rules={
      Array [
        Object {
          "min": 3,
          "required": true,
        },
      ]
    }
  >
    <Input
      type="text"
    />
  </FormItem>
  <FormItem
    label="Servings"
    name="servings"
    rules={
      Array [
        Object {
          "max": 10,
          "min": 1,
          "required": true,
          "type": "number",
        },
      ]
    }
  >
    <ForwardRef
      step={1}
    />
  </FormItem>
  <FormItem
    label="Ready in Min."
    name="readyInMinutes"
    rules={
      Array [
        Object {
          "max": 400,
          "min": 5,
          "required": true,
          "type": "number",
        },
      ]
    }
  >
    <ForwardRef
      step={1}
    />
  </FormItem>
  <FormItem
    label="Ingredients"
    name="ingredients"
    rules={
      Array [
        Object {
          "max": 20,
          "min": 1,
          "required": true,
          "type": "array",
        },
      ]
    }
  >
    <AutoCompleteForm
      error={false}
      form={[Function]}
      list={Array []}
      saveList={[Function]}
    />
  </FormItem>
  <FormItem
    label="Detailed Ingredients"
    name="detailedIngredients"
    rules={
      Array [
        Object {
          "max": 10,
          "min": 1,
          "required": true,
          "type": "array",
        },
      ]
    }
  >
    <ListInput
      detailedIngredients={Array []}
      form={[Function]}
      setDetailedIngredients={[Function]}
    />
    <List
      deleteItem={[Function]}
      list={Array []}
    />
  </FormItem>
  <FormItem
    label="Instructions"
    name="instructions"
    rules={
      Array [
        Object {
          "min": 10,
          "required": true,
        },
      ]
    }
  >
    <TextArea />
  </FormItem>
  <FormItem
    label="Image"
    name="image"
    rules={
      Array [
        Object {
          "required": true,
        },
      ]
    }
  >
    <ImageUpload
      form={[Function]}
      imageUrl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFYAAABWCAYAAABVVmH3AAAMSGlDQ1BJQ0MgUHJvZmlsZQAASImVVwdYU8kWnltSSWiBCEgJvQkiSJcSQosgIFWwEZJAQokhIYjYlWUVXLuIgA1dFVF0dQVkrahrXQR7f1hQUdbFgg2VNymg637vve+d75t7/5w55z8lc++dAUCnhieV5qK6AORJCmTxESGsCalpLNIjgAMMaAEjYMTjy6XsuLhoAGXw/nd5ew0gyvtlFyXXP+f/q+gJhHI+AEgcxBkCOT8P4l8BwEv4UlkBAEQfqLeeXiBV4kkQG8hgghBLlThLjUuUOEONK1U2ifEciHcBQKbxeLIsALSboZ5VyM+CPNo3IHaTCMQSAHTIEAfyRTwBxJEQj8jLm6bE0A44ZHzDk/U3zowhTh4vawira1EJOVQsl+byZvyf7fjfkperGIxhBwdNJIuMV9YM+3YjZ1qUEtMg7pFkxMRCrA/xe7FAZQ8xShUpIpPU9qgpX86BPQNMiN0EvNAoiE0hDpfkxkRr9BmZ4nAuxHCFoEXiAm6ixneRUB6WoOGskU2Ljx3EmTIOW+PbwJOp4irtTypyktga/hsiIXeQ/02xKDFFnTNGLRQnx0CsDTFTnpMQpbbBbIpFnJhBG5kiXpm/DcR+QklEiJofm5IpC4/X2Mvy5IP1YotEYm6MBlcViBIjNTy7+DxV/kYQNwsl7KRBHqF8QvRgLQJhaJi6dqxdKEnS1It1SgtC4jW+r6S5cRp7nCrMjVDqrSA2lRcmaHzxwAK4INX8eIy0IC5RnSeekc0bG6fOBy8C0YADQgELKODIANNANhC39TT1wF/qmXDAAzKQBYTARaMZ9EhRzUjgNQEUgz8hEgL5kF+IalYICqH+85BWfXUBmarZQpVHDngMcR6IArnwt0LlJRmKlgweQY34H9H5MNdcOJRz/9SxoSZao1EM8rJ0Bi2JYcRQYiâ€¦"
    />
  </FormItem>
  <FormItem
    label="Source Url"
    name="sourceUrl"
    rules={
      Array [
        Object {
          "required": false,
        },
      ]
    }
  >
    <Input
      addonBefore="http://"
      type="text"
    />
  </FormItem>
  <FormItem
    wrapperCol={
      Object {
        "offset": 10,
        "span": 12,
      }
    }
  >
    <Button
      block={false}
      className="mr-2"
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Submit
    </Button>
    <Button
      block={false}
      className="mr-2"
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
    >
      Reset
    </Button>
  </FormItem>
</ForwardRef(InternalForm)>
`;
